pallets/identity/rpc/src/lib.rs

```diff
@@ -0,0 +1,107 @@
+use codec::Codec;
+use jsonrpc_core::{Error as RpcError, ErrorCode, Result};
+use jsonrpc_derive::rpc;
+use polymesh_primitives::{IdentityId, Ticker};
+use polymesh_runtime_identity_rpc_runtime_api::IdentityApi as IdentityRuntimeApi;
+use sp_blockchain::HeaderBackend;
+use sp_runtime::{
+    generic::BlockId,
+    traits::{Block as BlockT, ProvideRuntimeApi},
+    Perbill,
+};
+use std::sync::Arc;
+
+#[rpc]
+pub trait IdentityApi<BlockHash, IdentityId, Ticker> {
+    /// Below function use to tell whether the given did has valid cdd claim or not
+    #[rpc(name = "identity_isIdentityHasValidCdd")]
+    fn is_identity_has_valid_cdd(
+        &self,
+        did: IdentityId,
+        buffer_time: Option<u64>,
+        at: Option<BlockHash>,
+    ) -> Result<(bool, Option<IdentityId>)>;
+
+    /// Below function is used to query the given ticker DID.
+    #[rpc(name = "identity_getAssetDid")]
+    fn get_asset_did(&self, ticker: Ticker, at: Option<BlockHash>) -> Result<Option<IdentityId>>;
+}
+
+/// A struct that implements the [`IdentityApi`].
+pub struct Identity<C, M> {
+    client: Arc<C>,
+    _marker: std::marker::PhantomData<M>,
+}
+
+impl<C, M> Identity<C, M> {
+    /// Create new `Identity` instance with the given reference to the client.
+    pub fn new(client: Arc<C>) -> Self {
+        Self {
+            client,
+            _marker: Default::default(),
+        }
+    }
+}
+
+/// Error type of this RPC api.
+pub enum Error {
+    /// The transaction was not decodable.
+    DecodeError,
+    /// The call to runtime failed.
+    RuntimeError,
+}
+
+impl From<Error> for i64 {
+    fn from(e: Error) -> i64 {
+        match e {
+            Error::RuntimeError => 1,
+            Error::DecodeError => 2,
+        }
+    }
+}
+
+impl<C, Block, IdentityId, Ticker> IdentityApi<<Block as BlockT>::Hash, IdentityId, Ticker>
+    for Identity<C, Block>
+where
+    Block: BlockT,
+    C: Send + Sync + 'static,
+    C: ProvideRuntimeApi,
+    C: HeaderBackend<Block>,
+    C::Api: IdentityRuntimeApi<Block, IdentityId, Ticker>,
+    IdentityId: Codec,
+    Ticker: Codec,
+{
+    fn is_identity_has_valid_cdd(
+        &self,
+        did: IdentityId,
+        buffer_time: Option<u64>,
+        at: Option<<Block as BlockT>::Hash>,
+    ) -> Result<(bool, Option<IdentityId>)> {
+        let api = self.client.runtime_api();
+        let at = BlockId::hash(at.unwrap_or_else(||
+            // If the block hash is not supplied assume the best block.
+            self.client.info().best_hash));
+        api.is_identity_has_valid_cdd(&at, did, buffer_time)
```

Comment:
After review `is_identity_has_valid_kyc` (which is used by `is_identity_has_valid_cdd`), we have two possible outputs:
- `(true, Some(...))` and,
- `(false, None)`
It is not related with your changes @satyamakgec, but it would nice if you could clean this too ;) . 
I think this function could return just an `Option<IdentityId>`.
What do you think?